diff_drive_controller/src/velocity_limiter.cpp

```diff
@@ -0,0 +1,146 @@
+/*
+ * velocity_limiter.cpp
+ *
+ *  Created on: May 30, 2019
+ *      Author: christof
+ */
+#include <diff_drive_controller/velocity_limiter.hpp>
+#include <cassert>
+
+namespace diff_drive_controller {
+
+VelocityLimiter::VelocityLimiter()
+        : _wheel_separation(0.0f)
+        , _vel_x_max(0.0f)
+        , _mass(1.0f)
+        , _inertia(0.0f)
+        , _f_wheel_max(0.0f)
+        , _f_wheel_min(0.0f)
+        , _has_velocity_limits(false)
+        , _has_acceleration_limits(false){};
+
+bool VelocityLimiter::init(float wheel_separation, bool has_velocity_limits, bool has_acceleration_limits,
+        float vel_x_max, float vel_th_max, float acc_x_max, float acc_x_min, float acc_th_max) {
+    if (wheel_separation <= 0.0f) {
+        return false;
+    }
+    _wheel_separation = wheel_separation;
+    _has_velocity_limits = has_velocity_limits;
+    _has_acceleration_limits = has_acceleration_limits;
+
+    if (_has_acceleration_limits) {
+        if (acc_x_max <= 0 || acc_x_min >= 0) {
+            return false;
+        }
+
+        _f_wheel_max = 0.5f * _mass * acc_x_max;
+        _f_wheel_min = 0.5f * _mass * acc_x_min;
+
+        // J * acc_th_max = 2 * f_wheel_max / (wheel_separation / 2)
+        _inertia = _wheel_separation * _f_wheel_max / acc_th_max;
```

Comment:
Hmm, should probably put a warning somewhere saying the cap for angular acceleration implicitly defines the moment of inertia, so it must be set to the physical limits of the hardware. By default I'd probably assume a uniform cylindrical mass distribution assumption instead.